src/main/resources/rules/crime/means/calculations/calculation.drl

```diff
@@ -16,71 +17,282 @@ import com.laa.model.OtherIncome;
 import com.laa.model.Partner;
 import com.laa.model.civil.enums.PaymentFrequency;
 import com.laa.model.civil.enums.IncomeType;
+import com.laa.model.civil.enums.OutgoingType;
 import java.util.Optional;
 import com.laa.model.Dependent;
+import com.laa.model.Outgoing;
 import com.laa.model.civil.enums.RelationToApplicant;
 import java.util.ArrayList;
 import com.laa.model.Dependent;
 import com.laa.model.enums.AccountType;
 
-
-rule "set courtType and caseType"
-lock-on-active  true
+rule "set courtType"
 	when
-		$case : CrimeCase(  courtType   != null, caseType != null)
-		$decision : CrimeMeansDecisionReport(courtType == null , caseType == null)
+		$case : CrimeCase(  courtType != null)
+		$decision : CrimeMeansDecisionReport(courtType == null)
 	then
 		$decision.setCourtType($case.getCourtType());
+		update($decision);
+end
+
+
+rule "set caseType"
+	when
+		$case : CrimeCase(caseType != null)
+		$decision : CrimeMeansDecisionReport(caseType == null)
+	then
 		$decision.setCaseType($case.getCaseType());
 		update($decision);
 end
 
 
-rule "calculate employed income and gross household income"
+rule "set applicant's employment status"
+lock-on-active  true
+	when
+		$case : CrimeCase( )
+		$decision : CrimeMeansDecisionReport( employmentStatus == null  )
+		
+	then
+	
+	 if ($case.getMeansInformation().getApplicant().getEmployedIncomeHistory() == null || $case.getMeansInformation().getApplicant().getEmployedIncomeHistory().isEmpty()) {
+	 		$decision.setEmploymentStatus(EmploymentStatus.UNEMPLOYED);
+	 } 
+	 else {
+	 		$decision.setEmploymentStatus(EmploymentStatus.EMPLOYED);
+	 }
+	 
+		update($decision);
+end
+
+// frequency for tax and ni could not be same as income frequecy, so model may change, for now it kept same.
+
+rule "calculate applicant's employed income,ni, tax , gross household income"
```

Comment:
this rule is too big, it might be a good idea to split the rules up, one to process monthly emplyed income and the other to process weekly employed income, and you can do a check  in the when cause for monthly or weekly